test/test_conversion_driver_esm1p5.py

```diff
@@ -80,11 +80,16 @@ def test_get_nc_write_path_unrecognized_unit():
     nc_write_dir = Path("netCDF")
     expected_nc_write_path = nc_write_dir / f"aiihca.p{unknown_key}-005007.nc"
 
-    nc_write_path = esm1p5_convert.get_nc_write_path(
-                        Path(ff_name),
-                        nc_write_dir,
-                        (ff_year, ff_month, 1)
-                    )
+    with pytest.warns(RuntimeWarning) as record:
+        nc_write_path = esm1p5_convert.get_nc_write_path(
+                            Path(ff_name),
+                            nc_write_dir,
+                            (ff_year, ff_month, 1)
+                        )
+        if not record:
+            msg = ("No warning with unknown filename key "
+                   f"'{unknown_key}' in '{ff_name}'")
+            pytest.fail(msg)
```

Comment:
What is the purpose of L89-92?
the 
```py
with pytest.warns(....):
    ...
```
block will already fail if the warning is not raised. 
I think this could be simplified to:
```suggestion
    with pytest.warns(RuntimeWarning):
        nc_write_path = esm1p5_convert.get_nc_write_path(
                            Path(ff_name),
                            nc_write_dir,
                            (ff_year, ff_month, 1)
                        )
```

Reply:
I think this suggestion makes sense. This was based on a previous example, where an additional message was used to provide useful detail to the user on why a test failed. In this case, the test is probably simple enough that its self explanatory and doesn't need additional information.